client/client_update.go

```diff
@@ -245,7 +246,7 @@ func processUpdateResponse(response *http.Response) (interface{}, error) {
 
 	case http.StatusNoContent:
 		log.Debug("No update available")
-		return nil, nil
+		return nil, ErrNoDeploymentAvailable
```

Comment:
Without having investigated anything at all, I have a hunch that if you're getting test failures, this could be a quite likely source of changes in behavior.